packaging/nix/package.nix

```diff
@@ -0,0 +1,44 @@
+{
+  # Dependencies:
+  lib,
+  buildFHSEnv,
+  umu-launcher-unwrapped,
+  # Public API:
+  version ? null,
+  withTruststore ? args.truststore or true,
+  withDeltaUpdates ? true,
+  # Deprecated:
+  # NOTE: this relies on there not being a `pkgs.truststore` or `pkgs.cbor2`...
+  # If there was, callPackage would provide them, causing our warnings to false-positive.
```

Comment:
I'm not sure what this means exactly. Why wouldn't they be there?

Reply:
> I'm not sure what this means exactly. 

If `pkgs.truststore` or `pkgs.cbor2` are ever added to nixpkgs, the warnings below will erroneously trigger.

Remember, `callPackage` automatically supplies attrs from `pkgs` to the package-function's args.

The warnings below are triggered by these deprecated args having been provided: e.g. `warnIf (args?truststore)`.

> Why wouldn't they be there?

Because there are no packages with those names. Luckily, for now.

The note clarifies we are assuming (and relying) on there **not** being `pkgs.truststore` or `pkgs.cbor2` attrs.

I.e. we're relying on `callPackage` **not** automatically supplying these args for us, because we want to check if a **user** supplied deprecated args.

Reply:
Hmm okay

Reply:
I've pushed some (hopefully clearer) new wording...

I also added datestamps & a link to this PR next to each deprecation.